plugins/evaluation/__init__.py

```diff
@@ -212,6 +230,56 @@ def _get_evaluation_type(view, pred_field):
     return label_type, eval_type, methods
 
 
+def _add_custom_metrics(ctx, inputs, eval_type, method):
+    supported_metrics = []
+    for operator in foo.list_operators(type="operator"):
```

Comment:
This really should just be baked in... having the contract defined here is not maintainable. Something like

```python
foo.list_operators_with_metadata(type="operator", eval_type=eval_type, method=method)
```

Reply:
Although the logic here should be colocated with the custom metric implementation, so that part is fine for now.

Reply:
yes I agree, was just getting something working to start.

More generally, we should have a think about the concept of "operator classes" like `EvaluationMetric(Operator)` introduced here and how we want that to work holistically. For example, should the concept of `tags` be in `fiftyone.yml` so that we can access information about the type of operators without having to instantiate them.